here-naksha-lib-psql/src/test/java/com/here/naksha/lib/psql/PsqlStorageTests.java

```diff
@@ -881,6 +893,77 @@ void intersectionSearch() throws NoCursor {
     }
   }
 
+
+  @Test
+  @Order(112)
+  @EnabledIf("runTest")
+  void notIndexedPropertyRead() throws NoCursor, IOException {
+    assertNotNull(storage);
+    assertNotNull(session);
+
+    // given
+    final String jsonReference = "[{\"id\":\"urn:here::here:Topology:106003684\",\"type\":\"Topology\",\"prop\":{\"a\":1}}]";
+    final WriteXyzFeatures request = new WriteXyzFeatures(collectionId());
+    final XyzFeature feature = new XyzFeature("featureWithExtraProperty_id");
+    feature.setGeometry(new XyzPoint(4.0d, 5.0));
+    feature.getProperties().put("color", "red");
+    feature.getProperties().put("weight", 60);
+    feature.getProperties().put("ids", new Integer[] {2,1,9});
+    ObjectReader reader = Json.get().reader();
+    feature.getProperties().put("references", reader.readValue(jsonReference, Object.class));
+
+    request.add(EWriteOp.CREATE, feature);
+    try (final ForwardCursor<XyzFeature, XyzFeatureCodec> cursor =
+             session.execute(request).getXyzFeatureCursor()) {
+      assertTrue(cursor.next());
+    } finally {
+      session.commit(true);
+    }
+
+    Consumer<ReadFeatures> expect = readFeaturesReq -> {
+      try (final ForwardCursor<XyzFeature, XyzFeatureCodec> cursor =
+               session.execute(readFeaturesReq).getXyzFeatureCursor()) {
+        cursor.next();
+        assertEquals("featureWithExtraProperty_id", cursor.getId());
+      } catch (NoCursor e) {
+        throw unchecked(e);
+      }
+    };
+
+
+    // when - search for int value
+    ReadFeatures readFeatures = new ReadFeatures(collectionId());
+    POp weightSearch = POp.eq(new NonIndexedPRef("properties", "weight"), 60);
+    readFeatures.setPropertyOp(weightSearch);
+    // then
+    expect.accept(readFeatures);
+
+    // when - search null value
+    POp exSearch = POp.isNotNull(new NonIndexedPRef("properties", "color"));
+    readFeatures.setPropertyOp(exSearch);
+    // then
+    expect.accept(readFeatures);
+
+    // when - search array contains
+    POp arraySearch = POp.contains(new NonIndexedPRef("properties", "ids"), 9);
+    readFeatures.setPropertyOp(arraySearch);
+    // then
+    expect.accept(readFeatures);
+
+    // when - search by json object
```

Comment:
I also suggest adding one test where left side of comparison is JSON object (and not array)

Reply:
done